packages/web/src/services/BrowserInstanceService.ts

```diff
@@ -0,0 +1,53 @@
+/*
+Copyright 2024 Splunk Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import { safelyGetSessionStorage, safelySetSessionStorage } from '../utils/storage';
+import { generateId } from '../utils';
+
+const BROWSER_INSTANCE_ID_KEY = 'browser_instance_id';
+
+/**
+ * BrowserInstanceService is responsible for generating and storing a unique ID for the tab.
+ * THe ID is stored in the session storage, so stays the same even after page reload
+ * as long as the tab stays on the same domain.
+ * This ID will be the same for all frames/context in the tab as long as they are on the same domain.
+ * Currently, this is simplified version which has this limitation:
+ *  - It does not cover the case when tab is duplicated -
+ *      browsers copy storage when duplicating tabs so the ID will be the same
+ *    To cover this case we need to implement a communication between tabs which requires asynchronous initialization.
+ *    This is not implemented yet as requires bigger refactoring.
+ */
+export class BrowserInstanceService {
+  static _id: string | undefined = undefined;
+
+  static get id(): string {
+    if(this._id) {
+      return this._id;
+    }
+
+
+    // Check if the ID is already stored in the session storage. It might be generated by another frame/context.
+    let browserInstanceId = safelyGetSessionStorage(BROWSER_INSTANCE_ID_KEY);
+    if(!browserInstanceId) {
+      browserInstanceId = generateId(64);
+      safelySetSessionStorage(BROWSER_INSTANCE_ID_KEY, browserInstanceId);
+    }
+    
+    this._id = browserInstanceId;
```

Comment:
if we didn't manage to persist it then wouldn't it be better to not attach it to spans? Because now let's say someone blocks session storage, and we see their session, it'll have a different browser.instance.id for each `scriptInstance`, tricking you into believing that these are separate windows.

Reply:
Good point. Please check https://github.com/signalfx/splunk-otel-js-web/pull/878/commits/861bed8b6fa8d6aa53ec419c6baa71fd5f7e7ab6